feature/dashboard/src/main/java/com/simprints/feature/dashboard/logout/syncdecline/LogoutSyncDeclineFragment.kt

```diff
@@ -62,7 +64,9 @@ class LogoutSyncDeclineFragment : Fragment(R.layout.fragment_logout_sync_decline
     }
 
     private fun processLogoutConfirmation() {
-        viewModel.logout()
-        findNavController().navigate(R.id.action_logoutSyncDeclineFragment_to_requestLoginFragment)
+        lifecycleScope.launch {
+            viewModel.logout().await()
```

Comment:
This feels a bit more complicated than it should. Couldn't the same be done with regular suspend functions and view model scope? Maybe with `runBlocking{}` to ensure it does wait for the execution?

Reply:
I already thought about using the  runblocking as it is simpler than using Deferred
But as you can see in the screenshot runblocking inside suspended methods is useless

![image](https://github.com/Simprints/Android-Simprints-ID/assets/6252874/8ea926b2-600d-4162-a567-e5b21ec3ed4b)

Reply:
Code blocks like the following would still lead to a race condition. 
```
logoutButton.setOnClickListener{
            logoutSyncViewModel.logout()
            findNavController().navigate(R.id.action_logoutSyncFragment_to_requestLoginFragment)
        }

```
------LogoutSyncViewModel-------------


```   fun logout() {
        viewModelScope.launch { logoutUseCase() }
    }
```
----------------LogoutUseCase-----------------
```suspend operator fun invoke() = runBlocking {
        // Cancel all background sync
        syncOrchestrator.cancelBackgroundWork()
        syncOrchestrator.deleteEventSyncInfo()
        // sign out the user
        authManager.signOut()
    }```